be/src/olap/storage_engine.cpp

```diff
@@ -428,6 +428,18 @@ std::vector<DataDir*> StorageEngine::get_stores_for_create_tablet(
     std::random_device rd;
     srand(rd());
     std::random_shuffle(stores.begin(), stores.end());
+    // Two random choices
+    for (int i = 0; i < stores.size(); i++) {
+        int j = i + 1;
+        if (j < stores.size()) {
+            if (stores[i]->tablet_set().size() > stores[j]->tablet_set().size()) {
+                std::swap(stores[i], stores[j]);
+            }
```

Comment:
@weizuo93 you can use `std::sort` directly, and `std::random_shuffle` is not needed?

Reply:
@acelyc111 
If we use std::sort directly, all tablets will be created on the same disk which has the least amount of tablets. I think it would be better to take into account both the number of tablets on the disk and the randomness of disk selection.

Reply:
@weizuo93 I got it.